closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1044 @@
+        }
+        return restricts.jdkSecureRandomProvider;
+    }
+
+    /**
+     * Restricted security mode secure random algorithm
+     *
+     * @return the secure random algorithm
+     */
+    public static String getRandomAlgorithm() {
+        if (!securityEnabled) {
+            printStackTraceAndExit(
+                    "Restricted security mode secure random algorithm can only be used when restricted security mode is enabled.");
+        }
+        return restricts.jdkSecureRandomAlgorithm;
+    }
+
+    /**
+     * FIPS mode will be enabled when the semeru.fips system property is true,
+     * or semeru.restrictedsecurity system property is set by using FIPS policy.
+     *
+     * @return true if FIPS is enabled
+     */
+    public static boolean isFIPSEnabled() {
+        return securityEnabled && (userSecurityNum == 1);
+    }
+
+    /**
+     * Check if the service is allowed in restricted security mode.
+     *
+     * @param service the service to check
+     * @return true if the service is allowed
+     */
+    public static boolean isServiceAllowed(Service service) {
+        if (securityEnabled) {
+            return restricts.isRestrictedServiceAllowed(service);
+        }
+        return true;
+    }
+
+    /**
+     * Check if the provider is allowed in restricted security mode.
+     *
+     * @param providerName the provider to check
+     * @return true if the provider is allowed
+     */
+    public static boolean isProviderAllowed(String providerName) {
+        if (securityEnabled) {
+            return restricts.isRestrictedProviderAllowed(providerName);
+        }
+        return true;
+    }
+
+    /**
+     * Check if the provider is allowed in restricted security mode.
+     *
+     * @param providerClazz the provider class to check
+     * @return true if the provider is allowed
+     */
+    public static boolean isProviderAllowed(Class<?> providerClazz) {
+        if (securityEnabled) {
+            return restricts.isRestrictedProviderAllowed(providerClazz);
+        }
+        return true;
+    }
+
+    /**
+     * Remove the security providers and only add the restricted security providers.
+     *
+     * @param props the java.security properties
+     * @return true if the restricted security properties loaded successfully
+     */
+    public static boolean configure(Properties props) {
+
+        // Check if restricted security is supported on this platform.
+        if ((userEnabledFIPS || userEnabledSecurity) && !isSecuritySupported) {
+            printStackTraceAndExit("Restricted security mode is not supported on this platform.");
+        }
+
+        try {
+            if (shouldEnableSecurity) {
+                if (debug != null) {
+                    debug.println("Restricted security mode detected, loading...");
+                }
+
+                // Read and set user restricted security settings.
+                initUserSetting();
```

Comment:
This should check that `restricts == null` (i.e. that `configure()` has not already been called.

Reply:
@keithc-ca 
I added a check when go into the “configure()” method, to check if the restricted security mode is already initialized or not.

Reply:
For the checks in "configure()", please see the latest comment.